Add detailed information on image component usage in comment section

anita-chouhan opened the pull request:
Fixes #7

anita-chouhan committed:
Add detailed information on image component usage in comment section

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
-  @param   lazyload     {bool}              Lazyload image or not
+  Generate, transform, manipulate, optimize and render images. 
+
+	What it can do:
```
_What it does_ might be more appropriate

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+	8. And finally, renders the image
+
+	How to use:
+  -----------
```
Dashes line is not aligned with the first character.

anita-chouhan replied:
Not showing in my editor.
<img width="960" alt="Screenshot 2022-06-23 at 5 43 23 PM" src="https://user-images.githubusercontent.com/30666052/175296036-d9518505-bee9-4c16-befd-8596f1d1419d.png">

rungta replied:
Maybe because spaces and tabs have been mixed up. We always use spaces in our code, not tabs. Make sure the correct whitespace character is being used. If you cannot see the difference in your editor, try turning on invisible characters.

anita-chouhan replied:
Thank you for the suggestion.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+
+	Configuration parameters:
+	@param   asset                {AssetFileModel}          [Required] Image asset to be rendered
+	@param   breakpoint_assets    {array|AssetFileModel}    Array of image assets to be rendered for different breakpoints
```
> Array of image assets to be rendered for different breakpoints

Just _Image assets to be rendered for different breakpoints_ is sufficient. Data type is already specified in the third column.

anita-chouhan replied:
Done.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+	@param   preset               {string}                  Pre-defined image transformation configuration name
+	@param   class                {string}                  Class attribute for the <img> tag
+	@param   alt                  {string}                  Alt attribute for the <img> tag
+	@param   lazyload             {bool}                    Whether to lazyload images (default: true)
```
Move this as the second last item in this list.

anita-chouhan replied:
Done.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+	@param   webp                 {bool}                    Whether to generate WebP versions (default: true)
+
+	Example of simple usage:
+  --------------
```
Misaligned dashes

anita-chouhan replied:
Not showing in my editor.

rungta replied:
Anita, clearly the bug is present in the code. If you cannot see it at your end, try something different (a different tool, or view the file on Github or the command line or something). Cannot stop at saying “not showing in my editor”.

anita-chouhan replied:
Fixed.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+  <figure>
+    {% include '_components/_image' with { params: {
+      asset: image,
+      class: 'profile-pic',
```
Let us use some Tailwind classes in the example.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+	Preset options:
+	---------------
+	@option   media          {string}      Any valid media query
+	@option   breakpoints    {array}       Array of transform values for different view breakpoints
```
_Array of_ can be dropped from the descriptions because it is already mentioned in the data type column.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+	@option   breakpoints    {array}       Array of transform values for different view breakpoints
+	@option   sizes          {array}       Array of size based transform values
+	@option   transform      {array}       Array of aspect and position related transform values
+	@option   guide          {string}      Any valid media query with size
```
Should mention that this will be rendered as the `sizes` attribute in the `<img>` tag.

anita-chouhan replied:
Done.

rungta commented on templates/_components/_image.twig:
````diff
@@ -3,13 +3,154 @@
+		```
+
+		HTML Output:
+		`<img src=”small.jpg” srcset=”small.jpg 20w, medium.jpg 30w, large.jpg 40w”>`
````
I think the file in the `src` attribute will probably be the largest generated transform, not the smallest. Please verify with this exact configuration and update the output.

rungta replied:
This will impact the `src` values in all the examples below.

anita-chouhan replied:
Tested and updated all `<img>` examples.

rungta commented on templates/_components/_image.twig:
```diff
@@ -3,13 +3,154 @@
+	{% include '_components/_image' with { params: {
+    asset: image,
+    breakpoint_assets: {
+     '--lg-plus': imageLarge,
```
Formatting.

rungta commented on templates/_components/_image.twig:
````diff
@@ -3,13 +3,154 @@
+
+	4) Art-directed images
+    ```
+    {% set queries = {
````
This should be in its own sub-section and explained just like presets.

anita-chouhan replied:
Added.

rungta commented on templates/_components/_image.twig:
````diff
@@ -3,13 +3,154 @@
+    ```
+		
+		HTML Output:
+		`<img src=”small.png” sizes=”734px” srcset=”small.png 400w, medium.png 600w, large.png 900w”>`
````
This example is not accurate. The `<source>` tags are missing.

anita-chouhan committed:
Language modifications for comment section

rungta commented on templates/_components/_image.twig:
```diff
@@ -63,9 +63,9 @@
-	@option   sizes          {array}       Array of size based transform values
-	@option   transform      {array}       Array of aspect and position related transform values
+	@option   breakpoints    {array}       Transform values for different view breakpoints
+	@option   sizes          {array}       Transform values for size that will render as the `sizes` attribute in the `<img>` tag
```
This is incorrect. The `guide` param of presets is rendered as the `sizes` attribute of the `<img>` (or could even be `<source>`) tag. The `sizes` parameter is _not_ rendered as the `sizes` attribute.

anita-chouhan replied:
Fixed.

rungta commented on templates/_components/_image.twig:
```diff
@@ -22,12 +22,12 @@
 	@param   alt                  {string}                  Alt attribute for the <img> tag
-	@param   lazyload             {bool}                    Whether to lazyload images (default: true)
 	@param   background           {string}                  Class name for background image
+  @param   lazyload             {bool}                    Whether to lazyload images (default: true)
```
Misaligned.

anita-chouhan replied:
Fixed.

rungta commented on templates/_components/_image.twig:
```diff
@@ -5,7 +5,7 @@
 
-	What it can do:
+	What it does:
 	---------------
```
Remove the colon at the end (it is already a heading) and reduce the dashes `---` to end where the heading text ends.

anita-chouhan replied:
Fixed.

anita-chouhan committed:
Format text and align dashes

anita-chouhan committed:
Fix the image rendering examples

anita-chouhan committed:
Add `queries` definition and examples to comment section

rungta committed:
Avoid lines longer than 80 chars, improve formatting

rungta committed:
Fix attribute quotes

rungta committed:
Improve formatting of `queries` declarations

rungta commented on templates/_components/_image.twig:
```diff
@@ -2,14 +2,266 @@
+          width="800"
+          height="889"
+          media="(min-width: 1200px)"
+          sizes="576px"
```
Why is the value of `sizes` set to `567px` in this case? We did not specify a `guide` param in our preset, did we?

anita-chouhan replied:
Removed `sizes`.

anita-chouhan committed:
Remove extra values from example image tags

anita-chouhan committed:
Merge branch 'mod/image-component-comment' of https://github.com/miranj/craft-boilerplate into mod/image-component-comment

anita-chouhan committed:
Format comment example text

anita-chouhan commented on templates/_components/_image.twig:
```diff
@@ -2,14 +2,266 @@
+          srcset="small.png 500w, medium.png 600w, large.png 700w, x-large.png 800w">
+        <img
+          src="large.png"
+          width
```
Removed.

anita-chouhan commented on templates/_components/_image.twig:
```diff
@@ -2,14 +2,266 @@
+          src="large.png"
+          width
+          height
+          sizes="576px"
```
Removed `sizes`.

rungta committed:
Re-structure & reformat docs for better flow

- Move all examples in a single place
- Use h1 and h2 for sectioning
- Improve Preset and Queries docs
- Improve param and option descriptions
- Add `lazyload: true` example
- Add `webp: true` example
- Improve & fix docs code formatting issues
